template/src/pageComponents/buy-me-coffee/ContractDemo.tsx

```diff
@@ -1,16 +1,34 @@
-import { useCallback } from 'react';
+import { useCallback, useMemo, useState } from 'react';
 import { clsx } from 'clsx';
 import useOnchainCoffeeMemos from '../../hooks/useOnchainCoffeeMemos';
-import FormBuyCoffee from './FormBuyCoffee';
 import Memos from './Memos';
+import BuyCoffeeFormStep from './steps/BuyCoffeeFormStep/BuyCoffeeFormStep';
+import StartTransactionStep from './steps/StartTransactionStep/StartTransactionStep';
+import TransactionCompleteStep from './steps/TransactionCompleteStep/TransactionCompleteStep';
 
 export default function BuyMeCoffeeContractDemo() {
+  const [transactionStep, setTransactionStep] = useState<string | null>(null);
+
   const { memos, refetchMemos } = useOnchainCoffeeMemos();
 
   const handleOncomplete = useCallback(() => {
     void refetchMemos();
   }, [refetchMemos]);
 
+  const asideContent = useMemo(() => {
+    if (transactionStep === 'START_TRANSACTION') {
```

Comment:
Maybe not something for this PR, but it's worth to think how we can simplify the understanding and re-use of those transaction step.

If you think about the majority of folks will look to BOAT as example on how to do **good** web3 transaction.

So when you code into BOAT, also take the opportunity to define best-practices that can be taught across the entire web3 ecosystem.

Reply:
Good point. I refactored this to use an enum instead, which is cleaner.